client/client.go

```diff
@@ -61,7 +61,11 @@ func (c *Client) SendRequest(method string, path string, payload interface{}, st
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return "", "", resp.StatusCode, err
+		if resp != nil {
+			return "", "", resp.StatusCode, err
+		} else {
+			return "", "", 502, err
```

Comment:
NIT:

```suggestion
			return "", "", http. StatusBadGateway, err
```

Instead of "just using" a bad gateway status code (which may or may not represent the problem) I'm thinking about just returning a zero value instead.